model/import/Report/ReportBuilder.php

```diff
@@ -18,49 +18,198 @@
+        $report->setData($resource ?? []);
+
+        $onlyWarningReports = [];
+        $warningAndErrorReports = [];
+
+        /** @var WarningValidationException[]|ErrorValidationException[] $exceptions */
+        foreach ($results->getErrorsAndWarnings() as $lineNumber => $exceptions) {
+            $warningReports = [];
+            $errorReports = [];
+
+            foreach ($exceptions as $exception) {
+                if ($exception instanceof WarningValidationException) {
+                    $warningReports[] = $this->createReportByException($exception);
+                }
+
+                if ($exception instanceof ErrorValidationException) {
+                    $errorReports[] = $this->createReportByException($exception);
+                }
+            }
+
+            if (empty($errorReports) && empty($warningReports)) {
+                continue;
+            }
+
+            $lineMainReport = Report::create(
+                empty($errorReports) ? Report::TYPE_WARNING : Report::TYPE_ERROR,
+                'line %s: %s',
+                [
+                    $lineNumber,
+                    '',
+                ]
+            );
+
+            foreach ($errorReports as $errorReport) {
+                $lineMainReport->add($errorReport);
+            }
+
+            foreach ($warningReports as $warningReport) {
+                $lineMainReport->add($warningReport);
+            }
+
+            if (empty($errorReports)) {
+                $onlyWarningReports[$lineNumber] = $lineMainReport;
+            }
+
+            if (!empty($errorReports)) {
+                $warningAndErrorReports[$lineNumber] = $lineMainReport;
+            }
         }
 
-        $type = (0 == count($importerResults->getWarningReports()) && 0 === count(
-                $importerResults->getErrorReports()
-            )) ? Report::TYPE_SUCCESS : Report::TYPE_INFO;
+        if (!empty($warningAndErrorReports)) {
+            $this->createAndAddSubReport(
+                $report,
+                $warningAndErrorReports,
+                Report::TYPE_ERROR,
+                '%s line(s) contain(s) an error and cannot be imported'
+            );
+        }
+
+        if (!empty($onlyWarningReports)) {
+            $this->createAndAddSubReport(
+                $report,
+                $onlyWarningReports,
+                Report::TYPE_WARNING,
+                '%s line(s) are imported with warnings'
+            );
+        }
+
+        return $report;
+    }
 
-        $report = new Report($type, $outerHeader, $resource);
-        $report->add(Report::createSuccess($innerHeader, $resource));
+    public function buildByException(Throwable $exception): Report
+    {
+        $report = Report::create(Report::TYPE_ERROR, 'CSV import failed');
+        $report->add(
+            Report::create(
+                Report::TYPE_ERROR,
+                'An unexpected error occurred during the CSV import. The system returned the following error: `%s`',
```

Comment:
As we have a tool, which is responsible for collecting list of translatable phrases (`tao/scripts/taoTranslate.php`) it should be modified to be aware of such interpolated messages

Reply:
I agree and we created another task for this. Thanks :)